controllers/image_generator.go

```diff
@@ -225,200 +226,143 @@ func (r *ImageGenerator) createJobFromFile(jobFileName string) (*batchv1.Job, er
-	latestCM, err := r.getPeerPodsCM()
-	if latestCM == nil || err != nil {
-		igLogger.Info("error getting peer-pods ConfigMap", "err", err)
-		return false, ctrl.Result{Requeue: true}
-	}
 
-	if op == "create" && latestCM.Data[r.CMimageIDKey] != "" {
-		igLogger.Info("Image ID is already set, skipping image creation", "image-ID", latestCM.Data[r.CMimageIDKey])
+	if op == "create" && r.isImageIDSet() {
+		igLogger.Info("Image ID is already set, skipping image creation")
 		return true, ctrl.Result{Requeue: false}
 	}
 
-	if latestCM.Data == nil {
-		latestCM.Data = make(map[string]string)
-	}
-
-	// set CM with BOOT_FIPS if system is in FIPS mode
-	if op == "create" && latestCM.Data[fipsCMKey] == "" && r.fips {
-		igLogger.Info("setting BOOT_FIPS in peer-pods ConfigMap")
-		latestCM.Data[fipsCMKey] = "true"
-		err = r.client.Update(context.TODO(), latestCM)
-		if err != nil {
-			igLogger.Info("error updating peer-pods ConfigMap with FIPS", "err", err)
-			return false, ctrl.Result{Requeue: true}
-		}
-	}
+	// file format: osc-podvm-[create|delete]-job.yaml
+	filename := "osc-podvm-" + op + "-job.yaml"
 
-	// file format: [azure|aws]-[CVM|VM]-image-[create|delete]-job.yaml
-	VMtype := defaultVMType
-	filename := r.provider + "-" + VMtype + "-image-" + op + "-job.yaml"
-	job, err = r.createJobFromFile(filename)
+	job, err := r.createJobFromFile(filename)
 	if err != nil {
 		igLogger.Info("error creating image creation job from yaml file", "err", err)
 		return false, ctrl.Result{Requeue: true}
 	}
 
-	if job.Status.Active == 0 && job.Status.Succeeded == 0 && job.Status.Failed == 0 { // job hasn't started yet
-		igLogger.Info("JobStatus: Job hasn't started yet", "job name", job.Name)
-		return false, ctrl.Result{RequeueAfter: 60 * time.Second}
-	} else if job.Status.Failed > 0 { // job failed, delete job?
-		igLogger.Info("JobStatus: Job has failed, delete job, keep pod for tracking", "job name", job.Name)
-		if err := r.deleteJobFromFile(filename, true); err != nil {
-			igLogger.Info("error deleting the job definition from yaml file", "err", err, "filename", filename)
-		}
-		return false, ctrl.Result{Requeue: true}
-	} else if job.Status.Succeeded > 0 { // job completed, continue
-		igLogger.Info("JobStatus: Job has completed successfully, fetch ImageID and add set in CM", "job name", job.Name)
-	} else if job.Status.Active > 0 { // job is still running
-		igLogger.Info("JobStatus: Job is still running", "job name", job.Name)
-		return false, ctrl.Result{RequeueAfter: 60 * time.Second}
-	} else {
-		igLogger.Info("JobStatus: Unexpected Job Status!!!", "job name", job.Name)
-		if err := r.deleteJobFromFile(filename, true); err != nil {
-			igLogger.Info("error deleting the job definition from yaml file", "err", err, "filename", filename)
-		}
+	err = r.checkJobStatus(job.Name, job.Namespace)
+	if err != nil {
+		igLogger.Info("error checking job status", "err", err)
 		return false, ctrl.Result{Requeue: true}
```

Comment:
if job is still running we requeue normally? AFAIR there's some backoff mechanism, is it suitable for a job that we know takes 5-10min ? cc @pmores

Reply:
also if error happens, should the job be deleted? IIUC next iteration will fail due to job with the same job name no?

Reply:
yes there is an exponential backoff on error. I have updated the code